core/shared/src/main/scala/org/http4s/Uri.scala

```diff
@@ -560,8 +553,8 @@ object Uri extends UriPlatform {
           )
       }
 
-    implicit val http4sInstancesForPath: Order[Path] with Semigroup[Path] =
-      new Order[Path] with Semigroup[Path] {
+    implicit val http4sInstancesForPath: Order[Path] with Semigroup[Path] with Hash[Path] =
+      new Order[Path] with Semigroup[Path] with Hash[Path] {
```

Comment:
I think something like this will fix the bincompat:

```suggestion
    def http4sInstancesForPath: : Order[Path] with Semigroup[Path] = http4sInstancesForPathBinCompat
    implicit val http4sInstancesForPathBinCompat: Order[Path] with Semigroup[Path] with Hash[Path] =
      new Order[Path] with Semigroup[Path] with Hash[Path] {
```

Reply:
Nice trick!